src/main/java/org/sonar/plugins/groovy/jacoco/AbstractAnalyzer.java

```diff
@@ -0,0 +1,219 @@
+import org.sonar.api.component.ResourcePerspectives;
+import org.sonar.api.measures.CoverageMeasuresBuilder;
+import org.sonar.api.measures.Measure;
+import org.sonar.api.resources.Project;
+import org.sonar.api.resources.Resource;
+import org.sonar.api.resources.ResourceUtils;
+import org.sonar.api.scan.filesystem.ModuleFileSystem;
+import org.sonar.api.scan.filesystem.PathResolver;
+import org.sonar.api.utils.SonarException;
+
+import javax.annotation.CheckForNull;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class AbstractAnalyzer {
+
+  private final ResourcePerspectives perspectives;
+  private final ModuleFileSystem moduleFileSystem;
+  private final FileSystem fileSystem;
+  private final PathResolver pathResolver;
+
+  public AbstractAnalyzer(ResourcePerspectives perspectives, ModuleFileSystem moduleFileSystem, FileSystem fileSystem, PathResolver pathResolver) {
+    this.perspectives = perspectives;
+    this.moduleFileSystem = moduleFileSystem;
+    this.fileSystem = fileSystem;
+    this.pathResolver = pathResolver;
+  }
+
+  private Resource getResource(ISourceFileCoverage coverage, SensorContext context) {
+    String fileRelativePath = coverage.getPackageName() + "/" + coverage.getName();
+    Resource resourceInContext = getResource(fileRelativePath, context);
+    if (resourceInContext != null && ResourceUtils.isUnitTestClass(resourceInContext)) {
+      // Ignore unit tests
+      return null;
+    }
+    return resourceInContext;
+  }
+
+  @CheckForNull
+  private Resource getResource(String fileRelativePath, SensorContext context) {
+    for (File sourceDir : moduleFileSystem.sourceDirs()) {
+      String absolutePath = sourceDir.getAbsolutePath() + "/" + fileRelativePath;
+      InputFile groovyFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(absolutePath));
+      if (groovyFile != null) {
+        return context.getResource(groovyFile);
+      }
+    }
+    return null;
+  }
+
+  public final void analyse(Project project, SensorContext context) {
+    if (!atLeastOneBinaryDirectoryExists(project)) {
+      JaCoCoExtensions.LOG.warn("Project coverage is set to 0% since there is no directories with classes.");
+      return;
+    }
+    String path = getReportPath(project);
+    if (path == null) {
+      JaCoCoExtensions.LOG.warn("No jacoco coverage execution file found for project " + project.getName() + ".");
+      return;
+    }
+    File jacocoExecutionData = pathResolver.relativeFile(moduleFileSystem.baseDir(), path);
+
+    try {
+      readExecutionData(jacocoExecutionData, context);
+    } catch (IOException e) {
+      throw new SonarException(e);
+    }
+  }
+
+  private boolean atLeastOneBinaryDirectoryExists(Project project) {
+    List<File> binaryDirs = moduleFileSystem.binaryDirs();
```

Comment:
I am pretty sure this call is deprecated and under the hood referes to a deprecated property.